### PR TITLE
Login and Sign Up

### PR DESCRIPTION
**Pull request by OnsNader**
- [x] sign-in FE

- [x] Sign-up FE

- [x] Sign-in FE validation

- [x] sign-up FE validation

### CODE REVIEW

File: public/css/signUpIn.css [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+}
+@media (max-width: 360px) {
+    .login, .signup{
+        padding-top: 100px;
+   }
+   .box h2 {
+       margin: 20px 0;
+   }
+}
+ #usernameErr {
```
Comment (knanahassouna1) [inline public/css/signUpIn.css]: use div instead of span if you wants that block

File: src/controllers/signIn.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+exports.get = (req,res) => {
+    res.render("sign-in", {
+        title: "Sign-in", headerFound: false, footerFound: false, asideFound:false, style: ["signUpIn"]
```
Comment (knanahassouna1) [inline src/controllers/signIn.js]: require sign in , sign up js files

File: src/controllers/signUp.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+exports.get = (req,res) => {
+    res.render("sign-up", {
+        title: "Sign-up", headerFound: false, footerFound: false, asideFound:false, style: ["signUpIn"]
```
Comment (knanahassouna1) [inline src/controllers/signUp.js]: require sign in , sign up js files

File: src/views/sign-in.hbs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+       
+            <div>
+       
+                <button type="submit">Sign in</button>
+            </div>
+       
+            <a href="#">Sign up</a>
+       
+        </div>
+
```
Comment (knanahassouna1) [inline src/views/sign-in.hbs]: remove white lines

File: src/views/sign-up.hbs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+   
+        </div>
+
+        <div>
+                <button type="submit" id="button">Sign up</button>
+        </div>
+
+        <a href="#">sign in</a>
+        </div>
+    </section>
```
Comment (knanahassouna1) [inline src/views/sign-up.hbs]: again remove white lines

File: src/views/sign-in.hbs [hunk 6]
```diff
@@ -0,0 +1,47 @@
+<section class="login">
+
```
Comment (mohannadhanafi) [inline src/views/sign-in.hbs]: Remove white spaces

File: src/views/sign-up.hbs [hunk 7]
```diff
@@ -0,0 +1,46 @@
+ <section class="signup">
+
```
Comment (mohannadhanafi) [inline src/views/sign-up.hbs]: Also here just remove the white spaces in hbs files

File: src/controllers/signIn.js (hunk 2 above)
Comment (SalwaMugh) [reply src/controllers/signIn.js]: yeah ons , link the validation files to hbs !!!

File: npm-debug.log [hunk 8]
```diff
@@ -0,0 +1,45 @@
+0 info it worked if it ends with ok
```
Comment (MohammedYehia) [inline npm-debug.log]: Remove this file

Comment (MohammedYehia) [inline npm-debug.log]: remove this file

File: public/js/sign-up.js [hunk 9]
```diff
@@ -0,0 +1,66 @@
+const username = document.getElementById("user");
```
Comment (MohammedYehia) [inline public/js/sign-up.js]: check this [validation topic](https://developer.mozilla.org/en-US/docs/Learn/HTML/Forms/Form_validation#Validating_forms_using_JavaScript) for later projects

File: src/controllers/index.js [hunk 10]
```diff
@@ -1,19 +1,29 @@
...
+Router.get('/signUp', signUp.get);
+
+// Sign In Routes
+Router.get('/signIn', signIn.get);
 
 // Admin Routes
 Router.get('/admin', admin.get);
 
-// Article Routes
+// Articles Routes
```
Comment (MohammedYehia) [inline src/controllers/index.js]: fix typo Articles -> Article

File: public/js/sign-up.js [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+const email = document.getElementById("email");
+const password = document.getElementById("pass");
+const usernameErr = document.getElementById("usernameErr");
+const emailErr = document.getElementById("emailErr");
+const passErr = document.getElementById("passErr");
+const usernameDiv = document.getElementsByClassName("username")[0];
```
Comment (MohammedYehia) [inline public/js/sign-up.js]: declared but never used

File: src/views/sign-up.hbs [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+            </div>  
+        <div class="password">  
+            <label for="password">Password:</label>
+            <input type="password" name="password" placeholder="Enter your password" id="pass" required>
+            <div id="passErr" class="err"></div>
+        </div>
+        <div>
+                <button type="submit" id="button">Sign up</button>
+        </div>
+        <a href="#">sign in</a>
```
Comment (MohammedYehia) [inline src/views/sign-up.hbs]: add `/signin` to href

File: src/views/sign-in.hbs [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+            </div>       
+            <div class="password">      
+                <label for="password">Password:</label>
+                <input type="password" name="password" placeholder="Enter your password" id="password">
+                <div id="passErr" class="err"></div>
+            </div>      
+            <div>      
+                <button type="submit">Sign in</button>
+            </div>      
+            <a href="#">Sign up</a>
```
Comment (MohammedYehia) [inline src/views/sign-in.hbs]: add `/signup` to href

File: public/js/sign-up.js (hunk 9 above)
Comment (knanahassouna1) [reply public/js/sign-up.js]: okay , thanks

**Pull request merged**